File: 6. Array/Easy/02. SecondLargest.cpp
class Solution {
public:
    int secondLargestElement(vector<int>& nums) {
        long long largest = LLONG_MIN, secondLargest = LLONG_MIN;
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] > largest){
                secondLargest = largest;
                largest = nums[i];
            } else if(nums[i] < largest && nums[i] > secondLargest){
                secondLargest = nums[i];
            }
        }
        return secondLargest == LLONG_MIN ? -1 : secondLargest;
    }
};

File: 6. Array/Easy/02. SecondLargest.py
class Solution:
    def secondLargestElement(self, nums):
        largest = float('-inf')
        secondLargest = float('-inf')
        for num in nums:
            if num > largest:
                secondLargest = largest
                largest = num
            elif num < largest and num > secondLargest:
                secondLargest = num
        return -1 if secondLargest == float('-inf') else secondLargest